### ISSUE TITLE
Invalid python_requires in setup.py - ">=3.5.*" not allowed

### ISSUE THREAD

**Issue by ColinKennedy** [PyCQA/pylint#3555]
This is a duplicate of https://github.com/PyCQA/astroid/issues/779 (astroid and pylint have the same issue related to their setup.py file.

### Steps to reproduce
- Install pylint
```sh
python3.6 -m pip install pylint==2.5.0 --user
```

- Run this
```python
import os
from distlib import database


def main():
    """Run the main execution of the current script."""
    path = os.path.join(os.path.expanduser("~"), ".local", "lib", "python3.6", "site-packages")
    distribution = database.DistributionPath([path])
    distributions = list(distribution.get_distributions())


if __name__ == "__main__":
    main()
```

### Current behavior
- Raises this exception - `ValueError: '.*' not allowed for u'>=' constraints`

### Expected behavior
`distributions` should be a non-empty list, containing at least astroid 2.2.0 as one of its entries

### pylint --version output
```
pylint 2.5.0
astroid 2.5.0
Python 3.6.8 (default, Aug  7 2019, 17:28:10)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

**Comment by Pierre-Sassoulas** [comment]
Closing because it was done in #3739
